api/test/utils/update.js

```diff
@@ -0,0 +1,193 @@
+
+/**
+ * get report of update
+ * @returns {JSON} report
+ */
+const getReport = async () => {
+  let res;
+  try {
+    res = await chai.request(ezunpaywallURL).get('/update/report');
+  } catch (err) {
+    logger.error(`getReport: ${err}`);
+  }
+  return res?.body?.report;
+};
+
+/**
+ * check if index exit
+ * @param {string} name Name of index
+ * @returns {boolean} if exist
+ */
+const checkIfIndexExist = async (name) => {
+  let res;
+  try {
+    res = await client.indices.exists({
+      index: name,
+    });
+  } catch (err) {
+    logger.error(`indices.exists in checkIfIndexExist: ${err}`);
+  }
+  return res.body;
+};
+
+/**
+ * delete index if it exist
+ * @param {string} name Name of index
+ */
+const deleteIndex = async (name) => {
+  const exist = await checkIfIndexExist(name);
+  if (exist) {
+    try {
+      await client.indices.delete({
+        index: name,
+      });
+    } catch (err) {
+      logger.error(`deleteIndex: ${err}`);
+    }
+  }
+};
+
+/**
+ * create index if it doesn't exist
+ * @param {string} name Name of index
+ * @param {JSON} index index in JSON format
+ */
+const createIndex = async (name, index) => {
+  const exist = await checkIfIndexExist(name);
+  if (!exist) {
+    try {
+      await client.indices.create({
+        index: name,
+        body: index,
+      });
+    } catch (err) {
+      logger.error(`indices.create in createIndex: ${err}`);
+    }
+  }
+};
+
+/**
+ * count how many documents there are in an index
+ * @param {string} name Name of index
+ * @returns {number} number of document
+ */
+const countDocuments = async (name) => {
+  const exist = await checkIfIndexExist(name);
+  let data;
+  if (exist) {
+    try {
+      data = await client.count({
+        index: name,
+      });
+    } catch (err) {
+      logger.error(`countDocuments: ${err}`);
+    }
+  }
+  return data.body.count ? data.body.count : 0;
+};
+
+/**
+ * checks if an update process is being processed
+ * @returns {boolean} if in update
+ */
+const checkIfInUpdate = async () => {
+  let res = true;
+  try {
+    res = await chai.request(ezunpaywallURL).get('/update/status');
+  } catch (err) {
+    logger.error(`checkIfInUpdate : ${err}`);
+  }
+  return res?.body?.inUpdate;
+};
+
+/**
+ * get state of update
+ * @returns {JSON} state
+ */
+const getState = async () => {
+  let res;
+  try {
+    res = await chai.request(ezunpaywallURL).get('/update/state');
+  } catch (err) {
+    logger.error(`getState : ${err}`);
+  }
+  return res?.body?.state;
+};
+
+/**
+ * delete a snapshot in ezunpaywall
+ * @param {String} filename name of file needed to be delete on ezunpaywall
+ */
+const deleteSnapshot = async (filename) => {
+  const filePath = path.resolve(downloadDir, filename);
+  const fileExist = await fs.pathExists(filePath);
+  if (fileExist) {
+    try {
+      await fs.unlinkSync(filePath);
+    } catch (err) {
+      logger.error(`deleteSnapshot: ${err}`);
+    }
+  }
```

Comment:
`fs-extra` has a [remove](https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md) method which is the same as `unlink`, except that it doesn't throw if the file doesn't exist. Either way, you can just remove the file and ignore `ENOENT` errors.

```suggestion
  try {
    await fs.remove(filePath);
  } catch (err) {
    logger.error(`deleteSnapshot: ${err}`);
  }
```